src/app/modules/user/register/register.component.ts

```diff
@@ -0,0 +1,100 @@
+import { MessageService } from '@shared/services/message.service';
+import { FormGroup, FormControl, Validators, FormBuilder, AbstractControl } from '@angular/forms';
+import { Component, OnInit } from '@angular/core';
+import { userObject } from '@shared/interfaces/userObject';
+import { AuthService } from '@shared/services/auth.service';
+import { Router } from '@angular/router';
+
+
+
+@Component({
+  selector: 'app-register',
+  templateUrl: './register.component.html',
+  styleUrls: ['./register.component.scss']
+})
+
+
+export class RegisterComponent implements OnInit {
+  constructor(
+    private authService: AuthService,
+    private messageService: MessageService,
+    private formBuilder: FormBuilder,
+    private router: Router
+  ) { }
+
+  mouseOverRegister: boolean = false;
+  registerForm: FormGroup;
+
+  onSubmit(registerObject: userObject): void {
+    console.log(registerObject)
+    if (!this.registerForm.invalid) {
+      this.authService.userRegister(registerObject).subscribe(
+        res => {
+          this.messageService.handleSuccess('Registration successful');
```

Comment:
you can move this logic in the register method in the service

Reply:
Done